library/std/src/sys_common/wtf8.rs

```diff
@@ -195,15 +222,17 @@ impl Wtf8Buf {
                     let code_point = unsafe { CodePoint::from_u32_unchecked(surrogate as u32) };
                     // Skip the WTF-8 concatenation check,
                     // surrogate pairs are already decoded by decode_utf16
-                    string.push_code_point_unchecked(code_point)
+                    string.push_code_point_unchecked(code_point);
+                    // The string now contains an unpaired surrogate.
+                    string.is_known_utf8 = false;
```

Comment:
Can `push_code_point_unchecked` panic? If so, we should probably move the `is_known_utf8` assignment to the line above.

Reply:
It can at least oom, so I've now moved the `is_known_utf8` assignment to the line above.